js/store.js

```diff
@@ -29,31 +44,39 @@ const store = {
     return null;
   },
 
-  setFirstParty(param) {
-    /**
-     * @param object with hostname as key
-     * @todo validate the param.-
-     * @todo code to be updated in next PR
-     */
-    return this.set(param);
+  setFirstParty(hostname, data) {
+    if (!hostname) {
+      throw new Error('setFirstParty requires a valid hostname argument');
+    }
+
+    const websites = clone(this._websites);
+    websites[hostname] = data;
+    this._write(websites);
   },
 
-  async set(websites) {
-    /**
-      * @todo code to be updated in next PR
-    */
-    return await browser.storage.local.set({ websites });
+  setThirdParty(origin, target, data) {
```

Comment:
I agree with @princiya here in that origin isn't specific of which origin. If we wanted to be more specific we might change this to `firstPartyOrigin` and make the collection code use `document` to infer what the first party is... with the assumption that more avenues of data collection might be used later etc.
I'm fine to change `origin` to `firstPartyOrigin` in a follow up, perhaps also with a description of the arguments being that this will be part of this objects API.